### ISSUE TITLE
Problem visualizing uncommited changes

### ISSUE THREAD

**Issue by ElisaPiccin** [allegroai/clearml#287]
Hello everybody,

I cloned locally the clearml repository and slightly modified the pandas_reporting.py (under \examples\reporting) to make some testing and saved it as pandas_reporting_tutorial.py. 
I'm using the Free ClearML Hosted Service and previously configured it with clearml-init.
When I run pandas_reporting_tutorial.py and experiment get correctly created and my data saved under Results (tab Plots), but under Execution tab the Uncommited changes section says "No changes logged" (the file does not exists on git, it is entirely new).
![ClearML_Capture](https://user-images.githubusercontent.com/12595792/104722731-92ec3d80-572e-11eb-8ef4-91f5ae117b21.PNG)

How can I retrieve the pandas_reporting_tutorial.py script (suppose I am not the author of it)?

Thanks,
Elisa

**Comment by bmartinn** [comment]
Hi @ElisaPiccin 
I'm posting here the answer from the Slack channel, for better visibility.

Seems like you are running the exact code as in the git repo. It points you to the repository "https://github.com/allegroai/clearml" , the script "examples/reporting/pandas_reporting.py" and the specific commit id "34c41cfc8c3419e06cd4ac954e4b23034667c4d9"
You can directly access it with this URL template:
`https://github.com/allegroai/clearml/blob/<commit_id>/<workind_directory>/<script_path>`
Or in this case:
https://github.com/allegroai/clearml/blob/34c41cfc8c3419e06cd4ac954e4b23034667c4d9/examples/reporting/pandas_reporting.py

>  If I make some changes in the original file it gets tracked (picture below)... but what if I use a completely new and git-untracked py script?

Basically it will capture the output of "git diff" so if you have a new file but it is Not added, then no changes will appear.
That said, this is usually the exception, most files that are not added to the git should not be logged, hence the logic.
Anyhow the idea is that the Agent will be able to replicate your entire code base, without you having to worry about constantly committing your code (not that it is bad, but we all tend to forget)

**Comment by jollyjonson** [comment]
Hi all, thanks for the discussion above — I’d like to raise a related concern.

ClearML seemingly relies on `git diff` to track uncommitted changes. However, as noted, `git diff` **does not capture new, untracked files** (i.e., files that haven't been `git add`ed). This means that **entire modules or scripts can go completely unlogged** in an experiment, simply because they’re new and not staged. Specifically in a scenario where you're dealing with little data and not have to rely on agents.

From a reproducibility and auditability perspective, this feels like a significant blind spot. It's quite easy to run code that never gets logged or stored, which undermines ClearML’s otherwise strong tracking.

Would you generally be open to adding an option/feature to also track untracked files in a repository, or is this a conscious design decision that isn’t subject to change?

I’m happy to open a new issue if this one isn’t the right place for that discussion.

Thanks!

**Comment by patrickdasilva-clearml** [comment]
Hi @ElisaPiccin , we're closing this issue as stale since it's been created too long ago. If you think the issue is still relevant, please make sure the issue persists on a recent version of ClearML (ideally the latest) before re-opening the issue. We'll gladly look at it again.
